include/swift/AST/DiagnosticsSema.def

```diff
@@ -4098,6 +4098,14 @@ WARNING(class_super_not_usable_from_inline_warn,none,
         "should be '@usableFromInline' or public",
         (bool))
 
+WARNING(meta_metatype,none,
+        "meta-metatype is probably not what you want", ())
+NOTE(meta_metatype_silence,none,
+     "add parentheses to silence this warning", ())
+WARNING(metatype_self,none,
+        "self of a metatype produces a meta-metatype", ())
+NOTE(metatype_self_silence,none,
+     "add parentheses to silence this warning", ())
```

Comment:
There needs to be tests for all of these diagnostics and the associated fix-its.

Also, it would be great if you copied these diagnostics over to the diagnostic localization file (located at `localization/diagnostics/en.yaml`) as well.

Reply:
Okay. Fixed but:

1) There really ought to be a test that ensures that the "localized" English content doesn't drift.
2) Better still, why isn't the English "localization" auto generated?

Reply:
`en.yaml` isn't auto generated because we intend to remove the message strings from `DiagnosticsSema.def`. The goal is for `DiagnosticsSema.def` to only contain the diagnostic ID and argument types, and the `.yaml` files will contain the message strings so that we don't need to rebuild the compiler every time we want to tweak the wording of a message. It's in this inconvenient state right now because the transition is still a work in progress

Reply:
Interesting. I still wish we'd auto-generate the YAML until we're ready to flip the switch. The duplication is just asking for bugs. And yes, I am assuming that the YAML file WAS auto-generated at one point and we could have saved that script, but perhaps I'm assuming too much.